src/components/linter.ts

```diff
@@ -1,248 +1,66 @@
 import * as vscode from 'vscode'
-import * as path from 'path'
-import * as fs from 'fs'
-import * as os from 'os'
-import {ChildProcessWithoutNullStreams, spawn} from 'child_process'
-import {EOL} from 'os'
 
 import type {Extension} from '../main'
+import { ChkTeX } from './linterlib/chktex'
+import { LaCheck } from './linterlib/lacheck'
+export interface ILinter {
+    readonly linterDiagnostics: vscode.DiagnosticCollection,
+    lintRootFile(): void,
+    lintFile(document: vscode.TextDocument): void,
+    parseLog(log: string, filePath?: string): void
+}
 
 export class Linter {
-    private readonly extension: Extension
+    readonly #linters: {[key: string]: ILinter} = {}
```

Comment:
We had better add `#lacheck: ILinter | undefined` and `#chktex: ILinter | undefined` instead of `#linters`.

We should avoid using `object` to prevent prototype pollution. See jlelong/LaTeX-Workshop-security/issues/3.

Reply:
I agree. Just wondering which one is better, explicit variables on each linter or a `Map<string, ILinter>`. I am a little bit inclined on the latter, which can be (very trivially) more friendly to possible future linters? Any idea?

Reply:
Using a `Map` looks good to me. @tamuratak ?

Reply:
It is up to you.